keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -291,80 +686,149 @@ No, the default behavior remains the same.
+2. During a rolling update of a StatefulSet with the feature enabled:
+   - If the feature gate is disabled in the midst of the rolling update, the behavior of the StatefulSet will shift to the default rolling update mechanism (updating one pod at a time). Existing pods that were terminated as part of the rolling update when the feature was enabled will continue their termination process. However, the creation of new pods will adhere to the default rolling update strategy, potentially leading to a slower update process if `maxUnavailable` was originally set to a value greater than 1.
+
+To summarize, the impact on running StatefulSets during feature enablement and rollback is primarily observed in the rolling update behavior. StatefulSets that begin a rolling update with the feature enabled will utilize the `maxUnavailable` setting until a change in the feature gate status is detected, after which the behavior will revert to the default update strategy.
 
 ###### What specific metrics should inform a rollback?
-TODO when we reach Beta
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Metric Name: statefulset_unavailability_violation
+
+Description: This metric counts the number of times a StatefulSet exceeds its maxUnavailable threshold during a rolling update. This metric increases whenever a StatefulSet is processed and it's observed that spec.replicas - status.readyReplicas > maxUnavailable. The metric is labeled with namespace, statefulset, and a reason label, where reason could be exceededMaxUnavailable. This provides a clear indication of which StatefulSets are not complying with the defined maxUnavailable constraint, allowing for quick identification and remediation.
+
+Usage: Administrators and SREs can use this metric to monitor and alert on StatefulSet configurations that violate the maxUnavailable constraint. This metric is particularly useful for ensuring that the rolling updates are proceeding as expected and that no unexpected unavailability is occurring. This metric can be aggregated across multiple clusters for large-scale monitoring.
+
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
-Will be tested when graduating to Beta.
 
+No, but it will be tested manually before merging the PR.
 
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
 No
 
 ### Monitoring Requirements
 
 ###### How can an operator determine if the feature is in use by workloads?
+
 If their StatefulSet rollingUpdate section has the field maxUnavailable specified with
-a value different than 1.
-The below command should show maxUnavailable value:
+a value different than 1. While in alpha and beta, the feature-gate needs to be enabled.
+
+The command bellow should show the maxUnavailable value:
```

Comment:
I agree it's not ideal, but we allowed for that in the past (since we don't have much better way of doing that and we don't want to expose a metric for every newly added field).

So it's actually ok (although i agree it's imperfect).

Reply:
Keeping it as is then, let me know if that is a problem, and you have other suggestion :)